### DISCUSSION TITLE
Auto emit child particles of a particle node

### DISCUSSION THREAD

**Discussion by JohnnyRouddro** [godotengine/godot-proposals#8935] [Engine Core]
If child particle nodes auto emit in game and in editor, it becomes faster to create complex vfx.
For example, here are 3 individual particle effects:

![individual](https://github.com/godotengine/godot-proposals/assets/30183816/798f27a4-3bcd-43ae-a81a-e605e04faa57)

And here are they combined:

![combined](https://github.com/godotengine/godot-proposals/assets/30183816/9421a6de-1264-444a-9e37-b780cc656c28)

I am currently using this piece of code on the root node of the particles to achieve the result.

```
@tool
extends Node

func _set(property, value):
	if property == "emitting" and value:
		propagate_call("restart")
```

**Comment by Calinou** [comment]
Having this behavior work based on parent-children relationships would likely break existing projects, as this structure is likely being relied upon already but without the expectation that enabling **Emitting** on the parent will automatically do so on the child nodes.

**Reply by Xorblax** [reply]
It'd need some kind of separate toggle. It wouldn't be able to mesh with the existing subemitter property as that seems to act for each individual particle, not the whole system. But as it is now a script does the job well enough I think
